packages/keyring-eth-hd/src/index.ts

```diff
@@ -123,187 +168,423 @@ class HdKeyring {
     return appKeyAddress;
   }
 
-  // exportAccount should return a hex-encoded private key:
-  async exportAccount(address, opts = {}) {
-    const wallet = this._getWalletForAccount(address, opts);
-    return bytesToHex(wallet.privateKey);
+  /**
+   * Export the private key for a specific account.
+   *
+   * @param address - The address of the account.
+   * @param opts - The options for exporting the account.
+   * @param opts.withAppKeyOrigin - An optional string to export the account
+   * for a specific app origin.
+   * @returns The private key of the account.
+   */
+  async exportAccount(
+    address: string,
+    opts?: { withAppKeyOrigin: string },
+  ): Promise<string> {
+    const wallet = opts
+      ? this.#getWalletForAccount(address, opts)
+      : this.#getWalletForAccount(address);
+    const { privateKey } = wallet;
+    assert(
+      privateKey instanceof Uint8Array,
+      'Expected private key to be of type Uint8Array',
+    );
+    return bytesToHex(privateKey);
   }
 
-  // tx is an instance of the ethereumjs-transaction class.
-  async signTransaction(address, tx, opts = {}) {
-    const privKey = this._getPrivateKeyFor(address, opts);
-    const signedTx = tx.sign(privKey);
+  /**
+   * Sign a transaction using the private key of the specified account.
+   *
+   * @param address - The address of the account.
+   * @param tx - The transaction to sign.
+   * @param opts - The options for signing the transaction.
+   * @returns The signed transaction.
+   */
+  async signTransaction(
+    address: string,
+    tx: TypedTransaction,
+    opts = {},
+  ): Promise<TypedTransaction> {
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    const signedTx = tx.sign(Buffer.from(privKey));
     // Newer versions of Ethereumjs-tx are immutable and return a new tx object
-    return signedTx === undefined ? tx : signedTx;
+    return signedTx ?? tx;
   }
 
-  // For eth_sign, we need to sign arbitrary data:
-  async signMessage(address, data, opts = {}) {
+  /**
+   * Sign a message using the private key of the specified account.
+   *
+   * @param address - The address of the account.
+   * @param data - The data to sign.
+   * @param opts - The options for signing the message.
+   * @returns The signature of the message.
+   */
+  async signMessage(
+    address: string,
+    data: string,
+    opts: HDKeyringAccountSelectionOptions = {},
+  ): Promise<string> {
     assertIsHexString(data);
     const message = remove0x(data);
-    const privKey = this._getPrivateKeyFor(address, opts);
-    const msgSig = ecsign(Buffer.from(message, 'hex'), privKey);
-    const rawMsgSig = concatSig(msgSig.v, msgSig.r, msgSig.s);
+    const privKey = this.#getPrivateKeyFor(address, opts);
+    const msgSig = ecsign(Buffer.from(message, 'hex'), Buffer.from(privKey));
+    const rawMsgSig = concatSig(
+      // WARN: verify this cast to Buffer
+      msgSig.v as unknown as Buffer,
```

Comment:
It looks like `@metamask/eth-sig-util` only accepts a Buffer here, while `msgSig.v` seems to be a `bigint`

Reply:
Hmm, does that cause an issue? A bigint is not a Buffer.

Reply:
Ah, strange, okay.